woocommerce/woocommerce-android
# Update Payment Method On Cash Payments

Closes: #11148 


**Do Not Merge -** [FluxC needs to be merged first](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/3054) and the hashcode needs to be updated in this PR.

Note: Github keeps saying the build is failing, but it is actually not failing - it just complains about the 'do not merge' flag.

### Description

Before this PR, the app would leave the `payment_method` and `payment_method_title` fields empty when an order was marked as paid using `Cash`. However, this behavior is not consistent with the web and the iOS app which set the `payment_method` to `cod` and `payment_method_title` to whatever the user choses in `/wp-admin/admin.php?page=wc-settings&tab=checkout&section=cod`. This PR fixes this issue by propagating the desired payment method and payment method title to FluxC. 

The app relies on the PaymentGateway being cached, when it is not cached a default `Pay in Person` title is used to avoid an extra network request - this scenario should be rare as the payment gateway is automatically cached on app startup and stored in a persistent storage.

For more details about this issue and the fix refer to pdfdoF-5ai-p2.

### Steps to reproduce


1. Create an order in the app
2. Open detail of the order
3. Tap on 'Collect Payment'
4. Select Cash
5. Confirm
6. Notice the order is marked as completed, but the payment_method and payment_method_title are empty (not visible on the UI in the app)

### Testing information


1. Change COD title on [SITE_URL]/wp-admin/admin.php?page=wc-settings&tab=checkout&section=cod
2. Restart the app
3. Create an order in the app
4. Open detail of the order
5. Tap on 'Collect Payment'
6. Select Cash
7. Confirm
8. Notice the order is marked as completed and your custom payment method title is visible on the order detail.


P.S. I also tested the scenario when the Gateway is not cached using Flipper.

### Images/gif

| Before | After | After Custom Title |
|--|--|--|
| <img src="https://github.com/user-attachments/assets/f38ab71f-5a2f-4097-938d-26c7728d4d5e" width="250px" /> | <img src="https://github.com/user-attachments/assets/f318eb4d-b310-41cc-9656-be053f40a6d4" width="250px" /> | <img src="https://github.com/user-attachments/assets/718b1db8-9d1a-4c6e-b63b-ab842a98c96e" width="250px" /> |




148
